### PR TITLE
Allow specifying node image to boot in kind Config type

### PR DESCRIPTION
**Pull request by munnerz**
I opted to make this a field in the Config structure for now.

Do you think we're better having this as a flag to `create` instead? This seemed like a simple solution and in-keeping with other config options.

/cc @BenTheElder

### CODE REVIEW

File: kind/pkg/build/node_image.go [hunk 1]
```diff
@@ -41,7 +41,7 @@ type NodeImageBuildContext struct {
 
 // NewNodeImageBuildContext creates a new NodeImageBuildContext with
 // default configuration
-func NewNodeImageBuildContext(mode string) (ctx *NodeImageBuildContext, err error) {
+func NewNodeImageBuildContext(mode, imageName, baseImageName string) (ctx *NodeImageBuildContext, err error) {
```
Comment (munnerz) [inline kind/pkg/build/node_image.go]: We may want to consider a struct to hold these config options... I'll leave it up to you whether we do this as part of this PR @BenTheElder

Comment (BenTheElder) [reply kind/pkg/build/node_image.go]: I think 3 is probably still low enough that inline args is fine, if we get any more we can look at refactoring as we won't have downstream users of the library just yet, we'd break them when we move this out of t-i anyhow

**Comment by BenTheElder** [comment]
LGTM, let's also add a flag to `kind create` since image will be a very common setting

File: kind/cmd/kind/create/create.go [hunk 2]
```diff
@@ -45,6 +46,7 @@ func NewCommand() *cobra.Command {
 	}
 	cmd.Flags().StringVar(&flags.Name, "name", "1", "the cluster name")
 	cmd.Flags().StringVar(&flags.Config, "config", "", "path to create config file")
+	cmd.Flags().StringVar(&flags.ImageName, "image", "", "name of the image to use for booting the cluster. If this is non-empty, it will override the value specified in the --config file.")
```
Comment (BenTheElder) [inline kind/cmd/kind/create/create.go]: might make this a little shorter later, but /shrug for now, it's certainly clear :-)

**Pull request merged**